components/input/input-otp.component.ts

```diff
@@ -0,0 +1,229 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { KeyValuePipe, NgForOf } from '@angular/common';
+import {
+  booleanAttribute,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  forwardRef,
+  Input,
+  numberAttribute,
+  OnChanges,
+  QueryList,
+  SimpleChanges,
+  ViewChildren,
+  ViewEncapsulation
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormArray,
+  FormBuilder,
+  FormsModule,
+  NG_VALUE_ACCESSOR,
+  ReactiveFormsModule,
+  Validators
+} from '@angular/forms';
+import { takeUntil, tap } from 'rxjs/operators';
+
+import { NzDestroyService } from 'ng-zorro-antd/core/services';
+import { NzSafeAny, NzSizeLDSType, NzStatus, OnTouchedType } from 'ng-zorro-antd/core/types';
+
+import { NzInputDirective } from './input.directive';
+
+@Component({
+  selector: 'nz-input-otp',
+  exportAs: 'nzInputOtp',
+  preserveWhitespaces: false,
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    @for (item of otpArray.controls | keyvalue; track $index) {
+      <input
+        nz-input
+        class="ant-otp-input"
+        type="text"
+        maxlength="1"
+        size="1"
+        [nzSize]="nzSize"
+        [formControl]="$any(otpArray.at($index))"
+        [nzStatus]="nzStatus"
+        (input)="onInput($index, $event)"
+        (focus)="onFocus($event)"
+        (keydown)="onKeyDown($index, $event)"
+        (paste)="onPaste($index, $event)"
+        #otpInput
+      />
+    }
+  `,
+  host: {
+    class: 'ant-otp'
+  },
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => NzInputOtpComponent),
+      multi: true
+    },
+    NzDestroyService
+  ],
+  imports: [FormsModule, NgForOf, NzInputDirective, ReactiveFormsModule, KeyValuePipe],
+  standalone: true
+})
+export class NzInputOtpComponent implements ControlValueAccessor, OnChanges {
+  @ViewChildren('otpInput') otpInputs!: QueryList<ElementRef>;
+
+  @Input({ transform: numberAttribute }) nzLength: number = 6;
+  @Input() nzSize: NzSizeLDSType = 'default';
+  @Input({ transform: booleanAttribute }) disabled = false;
+  @Input() nzStatus: NzStatus = '';
+  @Input() nzFormatter: (value: string) => string = value => value;
+  @Input() nzMask: string | null = null;
+
+  protected otpArray!: FormArray;
+  private internalValue: string[] = [];
+  private onChangeCallback?: (_: NzSafeAny) => void;
+  onTouched: OnTouchedType = () => {};
+
+  constructor(
+    private readonly formBuilder: FormBuilder,
+    private readonly nzDestroyService: NzDestroyService
+  ) {
+    this.createFormArray();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['nzLength']?.currentValue && changes['nzLength'].currentValue !== changes['nzLength'].previousValue) {
```

Comment:
I think currentValue will never be equal to previousValue at the same time, so the comparison here is unnecessary, WDYT?

Reply:
You'r right, in this case it's not needed